### DISCUSSION TITLE
Cannot figure out GitHub release versioning for s6-overlay

### DISCUSSION THREAD

**Discussion by MrMeeb** [renovatebot/renovate#26768] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.140.12

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### What is your question?

Hello!

I have a Dockerfile that contains the following arg:
```
ARG S6_OVERLAY_VERSION=3.1.5.0
```

I want Renovate to find the new version of s6-overlay based on the GitHub releases. I cannot seem to figure out how to do this. Using this config:
```
{
    "customManagers": [
      {
        "customType": "regex",
        "fileMatch": ["Dockerfile"],
        "matchStrings": [
          "ARG S6_OVERLAY_VERSION=(?<currentValue>.*)"
        ],
        "datasourceTemplate": "github-releases",
        "depNameTemplate": "just-containers/s6-overlay"
      }
    ],
    "packageRules": [
      {
        "matchDatasources": ["github-releases"],
        "matchDepNames": ["just-containers/s6-overlay"],
        "extractVersion": "^v(?<version>.*)$"
      }
    ]
}
```

I can get Renovate to recognise the current version in the Dockerfile, but the `updates` segment is blank:
```
"regex": [
           {
             "deps": [
               {
                 "depName": "just-containers/s6-overlay",
                 "currentValue": "3.1.5.0",
                 "datasource": "github-releases",
                 "replaceString": "ARG S6_OVERLAY_VERSION=3.1.5.0",
                 "updates": [],
                 "packageName": "just-containers/s6-overlay",
                 "versioning": "semver-coerced",
                 "warnings": [],
                 "sourceUrl": "https://github.com/just-containers/s6-overlay",
                 "registryUrl": "https://github.com",
                 "currentVersion": "3.1.5.0",
                 "fixedVersion": "3.1.5.0"
               }
             ],
             "matchStrings": ["ARG S6_OVERLAY_VERSION=(?<currentValue>.*)"],
             "depNameTemplate": "just-containers/s6-overlay",
             "datasourceTemplate": "github-releases",
             "packageFile": "Dockerfile"
           }
         ]
```

I think I've come to the realisation that `3.1.5.0` is not semver or semver-coerced, so I tried the following regex versioning instead:

```
{
    "customManagers": [
      {
        "customType": "regex",
        "fileMatch": ["Dockerfile"],
        "matchStrings": [
          "ARG S6_OVERLAY_VERSION=(?<currentValue>.*)"
        ],
        "datasourceTemplate": "github-releases",
        "depNameTemplate": "just-containers/s6-overlay"
      }
    ],
    "packageRules": [
      {
        "matchDatasources": ["github-releases"],
        "matchDepNames": ["just-containers/s6-overlay"],
        "extractVersion": "^v(?<version>.*)$",
        "versioning": "regex: ^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)\\.(?<build>\\d+)$"
      }
    ]
}
```

When doing this, I get the error `Dependency just-containers/s6-overlay has unsupported/unversioned value 3.1.5.0 (versioning=regex: ^(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)\.(?<build>\d+)$)`, so at this point I'm stuck. Testing the regex is regex101 seems to correctly group the 4 digits into major, minor, patch and build.

I've tried stripping the 'v' from the GitHub release, as can be seen in my configs, but I've also tried without that and instead by adding a v to my ARG and adapting any regex. Nothing seems to make a difference, and I'm not able to glean anything from the logs given to me - there doesn't seem to be any details on the information Renovate is getting from the repo or how that's being parsed. 

Thank you!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Try using loose versioning

**Reply by MrMeeb** [reply]
Amazing, thank you. Still finding my way around Renovate and that did the trick perfectly

**Reply by MrMeeb** [reply]
I'm still curious why the regex approach didn't work though - are you able to comment on where it might have gone wrong?

**Reply by viceice** [reply]
you've a space after `regex:` , remove it 😉

**Reply by MrMeeb** [reply]
ahhh, oops!
